hydra-node/test/Hydra/BehaviorSpec.hs

```diff
@@ -309,3 +309,10 @@ withHydraNode nodeId snapshotStrategy connectToChain action = do
     let hn' = Network{broadcast = const $ pure ()}
     let node = HydraNode{eq, hn = hn', hh, oc = OnChain (const $ pure ()), sendResponse = atomically . putTMVar response, env}
     connectToChain node
+
+-- | A 'Tracer' that works in 'IOSim' monad.
+-- This tracer uses the 'Output' event which uses converts value traced to 'Dynamic'
+-- which requires 'Typeable' constraint. To retrieve the trace use 'selectTraceEventsDynamic'
+-- applied to the correct type.
+traceInIOSim :: Typeable a => Tracer (IOSim s) a
+traceInIOSim = Tracer $ \a -> traceM a
```

Comment:
instead of exporting the `Tracer` data constructor, I'd say we can define this function in `Hydra.Logging` anyways?

Reply:
So that's what I started doing yes, but this required adding `io-sim` as a dependency to the library code, which seemed to me to be a hint something was not quite right as there's probably no reason to use the actual implementation in production code. But I am fine putting it back if you want.